### PR TITLE
feat(GroupImportFixer): Ability to configure which type of imports should be grouped

### LINKED ISSUE #7220: Group imports only in functions
Given the example below, is it possible to group only the imports of functions?

```diff
<?php

use Illuminate\Filesystem\Filesystem;
use Illuminate\Support\Arr;
use Illuminate\Support\Str;


-use function Laravel\Prompts\confirm;
-use function Laravel\Prompts\multiselect;
-use function Laravel\Prompts\select;
+use function Laravel\Prompts\{confirm, multiselect, select};

class InstallCommand extends Command
{
    // ...
}
```

### PR DESCRIPTION
**Pull request by Wirone**
Fixes #7220

Some existing test cases were changed because after changing the imports manipulation approach (each type separately) the final order of imports have changed (also because I've changed how `$insertIndex` is determined, as it's safer to insert group import where first removed import was, not the last - see the test case with `\DontTouch::me()`). I think it does not matter, as this fixer is not about the ordering, but about grouping.

### CODE REVIEW

File: src/Fixer/Import/GroupImportFixer.php [hunk 1]
```diff
@@ -23,12 +27,19 @@
...
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
 
 /**
  * @author Volodymyr Kupriienko <vldmr.kuprienko@gmail.com>
+ * @author Greg Korba <greg@codito.dev>
  */
-final class GroupImportFixer extends AbstractFixer
+final class GroupImportFixer extends AbstractFixer implements ConfigurableFixerInterface
 {
+    public const GROUP_CLASS_LIKES = 'class-likes';
```
Comment (kubawerlos) [inline src/Fixer/Import/GroupImportFixer.php:39]: `private`? I see they are used in tests, but I'd use the strings directly in testd.

At least `@internal`, please.

**Review by kubawerlos — CHANGES_REQUESTED** [review]

File: src/Fixer/Import/GroupImportFixer.php (hunk 1 above)
Comment (Wirone) [reply src/Fixer/Import/GroupImportFixer.php:39]: I have mixed feelings about it, as I believe it's good to provide a contract for options and values, so people can use it in their configs and feel safe about the behaviour. I think we already merged some fixers with public constants.

Comment (kubawerlos) [reply src/Fixer/Import/GroupImportFixer.php:39]: > I believe it's good to provide a contract for options and values

I'm not against it, but if/when we do it, let's do it properly, e.g. have the constant in docs.

Comment (Wirone) [reply src/Fixer/Import/GroupImportFixer.php:39]: You mean in this PR, or in general?

Comment (kubawerlos) [reply src/Fixer/Import/GroupImportFixer.php:39]: In genera, of course.

**Review by kubawerlos — APPROVED** [review]

File: src/Fixer/Import/GroupImportFixer.php (hunk 1 above)
Comment (Wirone) [reply src/Fixer/Import/GroupImportFixer.php]: After some offline discussion we decided to mark these constants as internal, and try to refactor current codebase to extract new public contract that would provide these 3 import types for usage in any place of the code. Also, as it was pointed out, we currently have "classy" naming convention for class-like stuff (classes, interfaces, traits, enums) that are covered by standard `use` import, that's why one of the constants introduced in this PR was renamed (also with proper value).

**Pull request merged**